peps/pep-0729.rst

```diff
@@ -0,0 +1,414 @@
+  happened recently with `partial stubs in PEP
+  561 <https://discuss.python.org/t/pep-561-clarification-regarding-n/32875/27>`_,
+  somebody would make a PR to the typing specification to clarify the
+  spec, and then the Typing Council would decide on the spec change.
+
+The runtime :mod:`typing` module will continue to be maintained by the
+CPython core developer team. However, any changes to the runtime module that
+affect type checker behavior should be made in conjunction with a change
+to the specification (see below) and should be approved by the Typing Council.
+For example, in Python 3.11 the core developers added the new function
+:func:`typing.assert_type`. If the Typing Council had been in place, this
+change would require a matching change to the specification and approval
+by the Typing Council. On the other hand, Python 3.11 also added the
+:func:`typing.get_overloads` introspection helper. As this function does not
+affect type checker behavior, it would not require approval by the Typing
+Council.
+
+Relationship with type checkers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Typing Council has no direct authority over type checkers; it cannot
+force them to implement particular features or make behavior changes. Type
+checkers are incentivized to follow the specification set out by the Council
+because it allows them to take advantage of shared resources, such as
+libraries that expose typing information that follows the specification,
+the stub files in typeshed, the ``typing`` standard library module, and
+user documentation that covers the standard type system.
+Type checkers are free to extend the type system or deviate from the
+specification, but they should document such differences clearly.
+
+The fact that type checkers need to implement any decisions made by the
+Typing Council acts as a useful brake on the Council, ensuring that its
+decisions are conservative and well-considered. Individual type checkers
+remain free to innovate as they see fit, and successful innovations can
+be incorporated into the standard type system.
+
+Projects
+--------
+
+Here are some efforts a Typing Council would be responsible for.
+
+Conformance test suite
+^^^^^^^^^^^^^^^^^^^^^^
+
+A conformance test suite would provide machine checkable documentation for how
+type checkers should check Python code, accompanied by the results of major type
+checker implementations on the test suite. A rough sketch for what this could
+look like was `created by Shantanu <https://github.com/hauntsaninja/type_checker_consistency>`_.
+
+This would contain prescriptive tests from behavior prescribed by previous PEPs
+and descriptive tests that let us document behavior of existing implementations
+in areas that are not prescribed by any standard. These descriptions would be
+useful to inform efforts below and to identify areas of focus for
+standardization.
+
+Specification for the type system
```

Comment:
I also worry about the size of this task. E.g. seeing Kevin’s spec of just a part of the type system (arguably the easy part :-) makes me worry. Not that I have a better idea…